src/main/java/Duke.java

```diff
@@ -1,10 +1,51 @@
+import user.Ui;
+
+
+/** The main Duke class, that allows the user to save tasks. */
 public class Duke {
+
+    private final Ui ui;
+
+    /**
+     * Constructor for the Duke class.
+     */
+    public Duke() {
+        ui = new Ui();
+    }
+
+    /**
+     * The main function that starts Duke.
+     *
+     * @param args String inputs, but not used for Duke.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Duke duke = new Duke();
+        duke.ui.greet();
+        String userInput;
+        String[] result;
+        boolean finished = false;
+        while (!finished) {
```

Comment:
maybe can put all this in a method call run() and the main method just evoke run() method